tests/launchsecurity/tdx.go

```diff
@@ -0,0 +1,105 @@
+/*
+ * This file is part of the KubeVirt project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Copyright The KubeVirt Authors
+ *
+ */
+
+package launchsecurity
+
+import (
+	v1 "kubevirt.io/api/core/v1"
+
+	"kubevirt.io/kubevirt/pkg/libvmi"
+	libvmici "kubevirt.io/kubevirt/pkg/libvmi/cloudinit"
+	"kubevirt.io/kubevirt/pkg/virt-config/featuregate"
+	"kubevirt.io/kubevirt/tests/console"
+	"kubevirt.io/kubevirt/tests/decorators"
+	"kubevirt.io/kubevirt/tests/framework/kubevirt"
+	"kubevirt.io/kubevirt/tests/libkubevirt/config"
+	"kubevirt.io/kubevirt/tests/libvmops"
+
+	expect "github.com/google/goexpect"
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+
+	"kubevirt.io/client-go/kubecli"
+)
+
+var _ = Describe("[sig-compute]Intel TDX", decorators.TDX, decorators.SigCompute, decorators.RequiresAMD64, Serial, func() {
+	// I could use embedded data but it's not worth the effort at the moment
+	const cloudInitUserData = "#cloud-config\n" +
+		"user: fedora\n" +
+		"password: fedora\n" +
+		"chpasswd: { expire: False }\n" +
+		"ssh_pwauth: true\n"
+
+	newTDXFedora := func() *v1.VirtualMachineInstance {
+		// Configure libvirt to use TDX
+		// Use fedora image with support for TDX
+		// NewFedora() uses fedora39 which doesn't support TDX yet
+		tdxOptions := []libvmi.Option{
+			libvmi.WithTDX(),
+			libvmi.WithMemoryRequest("512Mi"),
+			libvmi.WithContainerDisk("rootdisk1", "quay.io/containerdisks/fedora"),
```

Comment:
We usually use `ContainerDiskFedoraTestTooling` through `NewFedora`. Is there a reason to use a new one?

Reply:
It's because ContainerDiskFedoraTestTooling is running Fedora39, which doesn't support TDX, so I have to use quay.io/containerdisks/fedora as the TDX VM guest OS